File: app/src/main/java/com/example/pokeapi/Retrofit2/ApiRest.java
package com.example.pokeapi.Retrofit2;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiRest {

    final private Urls urls;
    public ApiRest(){

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Utils.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        urls = retrofit.create(Urls.class);


    }

    public Urls getService(){
        return urls;
    }

}

File: app/src/main/java/com/example/pokeapi/Retrofit2/ApiRest.kt
package com.example.pokeapi.Retrofit2

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.create

class ApiRest() {
    private var service: Urls
    var Utils = Utils()

    init {

        val retrofit: Retrofit = Retrofit.Builder()
                .baseUrl(Utils.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

        service = retrofit.create(Urls::class.java)

    }

    fun getService(): Urls{
        return service
    }
}


